### PR TITLE
chore: make tsc happy :^)

### PR DESCRIPTION
**Pull request by ArjixWasTaken**
~~This will be a draft until I can further review my own code.~~
~~This was a rushed PR written at 03:00 in the morning :p~~

🌟 Ready for review !🌟

### CODE REVIEW

**Review by ArjixWasTaken — COMMENTED** [review]
I do not know all the business logic yet, so it is highly possible that I made a few mistakes regarding that.

File: src/main/events/helpers/generate-lutris-yaml.ts [hunk 1]
```diff
@@ -27,11 +27,10 @@ export const generateYML = (game: Game) => {
...
           task: {
-            executable: path.join(
-              game.downloadPath,
-              game.folderName,
-              "setup.exe"
-            ),
+            executable:
+              game.downloadPath &&
+              game.folderName &&
+              path.join(game.downloadPath, game.folderName, "setup.exe"),
```
[the remark this answers is not in the document]
Comment (lilezek) [reply src/main/events/helpers/generate-lutris-yaml.ts:33]: Well this changes the result, right? I would leave the original code with exclamations which is the closest to the original code.

File: src/main/events/misc/show-open-dialog.ts [hunk 2]
```diff
@@ -4,8 +4,10 @@ import { registerEvent } from "../register-event";
 
 const showOpenDialog = async (
   _event: Electron.IpcMainInvokeEvent,
-  options: Electron.OpenDialogOptions
-) => dialog.showOpenDialog(WindowManager.mainWindow, options);
+  options: Electron.OpenDialogOptions,
+) =>
+  WindowManager.mainWindow &&
+  dialog.showOpenDialog(WindowManager.mainWindow, options);
```
Comment (lilezek) [inline src/main/events/misc/show-open-dialog.ts:10]: If this is purely to make tsc happy I would just put an exclamation here.

File: src/main/services/repack-tracker/xatab.ts [hunk 3]
```diff
@@ -26,77 +26,76 @@ const formatXatabDate = (str: string) => {
...
     try {
-      const repack = await getXatabRepack(($a as HTMLAnchorElement).href);
-
-      repacks.push({
-        title: $a.textContent,
-        repacker: "Xatab",
-        ...repack,
-        page,
-      });
+      Object.assign(repack, await getXatabRepack(href));
```
Comment (lilezek) [inline src/main/services/repack-tracker/xatab.ts:78]: This is not equivalent to the original code. Repack is being assigned before page.

File: src/main/events/helpers/generate-lutris-yaml.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArjixWasTaken) [reply src/main/events/helpers/generate-lutris-yaml.ts:33]: But that would be dirty.
You would be lying to typescript in the chance that it is actually undefined.

File: src/main/services/repack-tracker/xatab.ts (hunk 3 above)
Comment (ArjixWasTaken) [reply src/main/services/repack-tracker/xatab.ts:78]: It is not equivalent, but does the order matter?
page is not defined by the repack either way.

File: src/main/events/misc/show-open-dialog.ts (hunk 2 above)
Comment (ArjixWasTaken) [reply src/main/events/misc/show-open-dialog.ts:10]: You are spot on

File: src/main/events/helpers/generate-lutris-yaml.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (ArjixWasTaken) [reply src/main/events/helpers/generate-lutris-yaml.ts]: reverted the change, instead I added `console.assert` which only logs to the console when the values are falsy

**Comment by ArjixWasTaken** [comment]
This is now ready for review :^)

**Comment by zamitto** [comment]
Closing as with the merge of #315 this has been fixed. Thanks anyway!

**Pull request closed without merging**